File: repository-local/src/main/java/com/andyanika/translator/repository/local/LocalRepositoryModule.java
package com.andyanika.translator.repository.local;

//import androidx.room.Room;
//import android.content.Context;
//import android.content.SharedPreferences;
//
//import com.andyanika.translator.common.interfaces.LocalRepository;
//import com.andyanika.translator.common.scopes.ApplicationScope;
//
//import javax.inject.Named;
//
//import dagger.Module;
//import dagger.Provides;
//import io.reactivex.rxjava3.core.Scheduler;
//
//@Module
//public class LocalRepositoryModule {
//    @Provides
//    @ApplicationScope
//    LocalRepository getDataSource(TranslatorDao dao, SharedPreferences preferences, ModelsAdapter adapter, @Named("io") Scheduler ioScheduler) {
//        return new LocalRepositoryImpl(dao, preferences, adapter, ioScheduler);
//    }
//
//    @Provides
//    @ApplicationScope
//    TranslatorDao provideDatabase(@Named("app") Context context) {
//        return Room.databaseBuilder(context, DatabaseTranslator.class, "words_database").build().translatorDao();
//    }
//
//    @Provides
//    @ApplicationScope
//    SharedPreferences providePreferences(@Named("app") Context ctx) {
//        return ctx.getSharedPreferences("config", Context.MODE_PRIVATE);
//    }
//}

File: repository-local/src/main/java/com/andyanika/translator/repository/local/di/koin/LocalRepositoryModule.kt
package com.andyanika.translator.repository.local.di.koin

import android.content.Context
import androidx.room.Room
import com.andyanika.translator.common.interfaces.LocalRepository
import com.andyanika.translator.repository.local.DatabaseTranslator
import com.andyanika.translator.repository.local.LocalRepositoryImpl
import com.andyanika.translator.repository.local.ModelsAdapter
import org.koin.core.qualifier.named
import org.koin.dsl.module
import org.koin.dsl.single

val localRepositoryModule = module {
    single<LocalRepository> { LocalRepositoryImpl(get(), get(), get(), get(named("io"))) }

    single {
        Room.databaseBuilder(get(), DatabaseTranslator::class.java, "words_database")
            .build()
            .translatorDao()
    }

    single {
        val ctx: Context = get()
        ctx.getSharedPreferences("config", Context.MODE_PRIVATE)
    }

    single<ModelsAdapter>()
}
